src/engine/naive_engine.cc

```diff
@@ -208,8 +210,7 @@ class NaiveEngine final : public Engine {
     for (auto var : mutable_vars) {
       ++var->version_;
     }
-    CHECK(req_completed)
-        << "NaiveEngine only support synchronize Push so far";
+    future.wait();
```

Comment:
With naive engine it does not really make a difference but generally speaking this wait should be before the `version_` increment, right?